File: client/test-upload.py
import requests
import string
import random
import os

base_url = "https://labassignment-production.up.railway.app/v1"
email = ''.join(random.choices(string.ascii_lowercase, k=10)) + "@test.com"

# 1. Register
requests.post(f"{base_url}/auth/register", json={"email": email, "password": "password123", "full_name": "Test User"})

# 2. Login
resp = requests.post(f"{base_url}/auth/login", json={"email": email, "password": "password123"})
token = resp.json().get("access_token")

if not token:
    print("Failed to login", resp.text)
    exit(1)

# 3. Create a 3MB file
with open("large.jpg", "wb") as f:
    f.write(os.urandom(3 * 1024 * 1024))

# 4. Upload
headers = {"Authorization": f"Bearer {token}"}
files = {"images": ("large.jpg", open("large.jpg", "rb"), "image/jpeg")}
data = {
    "title": "Camera",
    "description": "Good camera",
    "category": "other",
    "price": "989",
    "condition": "like_new"
}

resp = requests.post(f"{base_url}/gadgets/", headers=headers, data=data, files=files)
print(resp.status_code, resp.text)

File: client/test-upload.js
import fs from 'node:fs';

async function testUpload() {
  // 1. Register a test user
  const email = "test" + Date.now() + "@example.com";
  const registerResp = await fetch("https://labassignment-production.up.railway.app/v1/auth/register", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password: "password123", full_name: "Test User" })
  });
  if (!registerResp.ok) {
    throw new Error(`Registration failed: ${await registerResp.text()}`);
  }
  
  // 2. Login
  const loginResp = await fetch("https://labassignment-production.up.railway.app/v1/auth/login", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password: "password123" })
  });
  const loginData = await loginResp.json();
  const token = loginData.access_token;
  
  // 3. Create dummy file
  fs.writeFileSync('dummy.jpg', 'fake image content');
  
  // 4. Upload
  const formData = new FormData();
  formData.append('title', 'Test Gadget');
  formData.append('description', 'Test desc');
  formData.append('category', 'smartphones');
  formData.append('price', '99.99');
  formData.append('condition', 'new');
  
  const fileBlob = new Blob(['fake image content'], { type: 'image/jpeg' });
  formData.append('images', fileBlob, 'dummy.jpg');
  
  try {
    const uploadResp = await fetch("https://labassignment-production.up.railway.app/v1/gadgets/", {
      method: "POST",
      headers: { "Authorization": `Bearer ${token}` },
      body: formData
    });
    console.log("Status:", uploadResp.status);
    console.log("Text:", await uploadResp.text());
  } catch (err) {
    console.error("Fetch failed:", err);
  }
}

testUpload();
